internal/parsing/parse.go

```diff
@@ -161,6 +182,63 @@ func parseWith(file fs.File, parsers []Parser, groupNumber int, log *zap.Sugared
 	return &finalResults, nil
 }
 
+func checkIfTestIDsAreUnique(parser Parser, tests []v1.Test, cfg Config) (bool, error) {
+	var identityRecipe *v1.TestIdentityRecipe
+	uniqueTestIdentifiers := make(map[string]struct{})
+
+FRAMEWORKS:
+	for framework, frameworkParsers := range cfg.FrameworkParsers {
+		for _, frameworkParser := range frameworkParsers {
+			if reflect.TypeOf(frameworkParser) != reflect.TypeOf(parser) {
+				continue
+			}
+
+			// TODO: What about "other" frameworks (i.e. when we don't match here)
```

Comment:
We have a fallback in the identity recipes in Cloud - maybe we should maintain that same behavior?

Reply:
Whoops, I wanted to address that still...